core/src/main/kotlin/message/Message.kt

```diff
@@ -0,0 +1,132 @@
+package com.juul.mcumgr.message
+
+import com.fasterxml.jackson.annotation.JsonCreator
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.fasterxml.jackson.annotation.JsonValue
```

Comment:
Ideally, you'd decouple your serialization from your public API.

Essentially, it's be best to expose these data structures without the serialization annotations, then you have a "bridge" layer that copies data from your public API structure to your serialization specific objects.

This allows your serialization tool and data structure to change without breaking the public facing API. Also means your public API isn't directly driven by the underlying serialization structure. So you can expose more Kotlin idiomatic data structures than maybe the underlying firmware adheres to.

Reply:
I think I understand. Since we are using jackson I would want to create custom serializers for each request and response & add them to the ObjectMapper? That way the objects do not need to be annotated. Akin to a moshi adapter?

Reply:
Doing the conversion within adapters works. More importantly though it's about what is exposed on the public API surface.

Essentially you want something like:

```kotlin
data class PublicApiObject(
    val a: Foo,
    val b: Bar
)

internal SerializationObject(
    val y: Hello,
    val z: World
)
```

The `SerializationObject` can be annotated with serialization annotations if the serialization library uses them.
How you get from `PublicApiObject` to `SerializationObject` could either be manually or via serialization library adapters.

The ultimate goal being decoupling, whereas here, there structure of `PublicApiObject` can be whatever makes the most sense for a library consumer (and can be idiomatic to Kotlin types). It also can remain unchanged even if the underlying serialization structure changes; as the library would just update it's conversion internals if changes occurred.

Reply:
So I fiddled with custom serializers (adapters) and felt it was overly complex and did not "describe" the API like an annotated data class. 

Instead I used a jackson feature called a mixin. Basically substituting a non-annotated class with an annotated one. You then add the mapping between the two classes to the object mapper, and viola, it works.

The new serializations objects I called payloads and put in the serialization module under `Payload.kt`.

Reply:
Great! Really, whatever gets the job done to convert from your public API to your private serialization API (and allows your public API to have different structure / variable naming than your serialization API — if needed). 👍